util/a2e/atoe_utils.c

```diff
@@ -545,22 +539,29 @@ atoe_vsnprintf(char *str, size_t count, const char *fmt, va_list args)
 					/* ibm@8665
 					 * Add floating point support for dbl2str & flt2str
 					 */
-					char *b;
-					int len;
+					char *tempPattern;
+					size_t freeCap;
+					double argument;
+					int requiredSpace;
 
-					b = a2e((char *)pattern, fmt - pattern);
+					tempPattern = a2e((char *)pattern, fmt - pattern);
 
 					/* Extract a double from args, this works for both doubles
 					 * and floats,
 					 * NB if we use float for a single precision floating
 					 * point number the result is wrong.
 					 */
-					len = sprintf(this.buffer, b, va_arg(args, double));
-					free(b);
-					b = e2a_string(this.buffer);
-					strcpy(this.buffer, b);
-					free(b);
-					this.buffer += len;
+					freeCap = this.end > this.buffer ? this.end - this.buffer : 0;
+					argument = va_arg(args, double);
+					requiredSpace = snprintf(this.buffer, freeCap, tempPattern, argument);
+					free(tempPattern);
+					CheckRet(requiredSpace);
```

Comment:
Shouldn't this return ERROR value, if your buffer is not large enough? How does it work with the SPECS?

Reply:
No it should write as many characters as the `freeCap` allows it and estimate the rest.

Reply:
Okk, thanks for the clarification.